### PR TITLE
Add unique query parameter to remove duplicate tracks

### PR DESCRIPTION
**Pull request by setoelkahfi**
Hey @JeffreyCA,
I like this. Thank you. I add the ability to only show the unique tracks on the widget and some other changes :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,6 +57,18 @@ Example:
 
 ![Spotify recently played](https://spotify-recently-played-readme.vercel.app/api?user=jeffreyca16&width=600)
 
+### Unique tracks
+To show only uniqe tracks of the recently played list, pass the `unique` query parameter and set it to `true`, `1`, `on`, or `yes`.
```
Comment (JeffreyCA) [inline README.md]: uniqe => unique

File: pages/api/index.ts [hunk 2]
```diff
@@ -71,11 +88,19 @@ export default async (req: NextApiRequest, res: NextApiResponse): Promise<void>
...
         const username = await getUsername(tokens.accessToken);
-        const recentlyPlayed = await getRecentlyPlayed(count, tokens.accessToken);
+        const limit = uniqueTrack ? Constants.defaultUniqueTrackSearchLimit : count
+        let recentlyPlayedItems = await getRecentlyPlayed(limit, tokens.accessToken);
+
+        if (uniqueTrack) {
+            // Remove duplicate tracks...
+            recentlyPlayedItems = recentlyPlayedItems.filter((v, i, a) => a.findIndex((t) => t.track.id === v.track.id) === i);
+            // ... then only return the number of items that are requested.
+            recentlyPlayedItems.slice(0, count);
```
Comment (JeffreyCA) [inline pages/api/index.ts]: `slice()` returns a copy and doesn't modify in-place, so this should be `recentlyPlayedItems = ...`,

File: utils/Constants.ts [hunk 3]
```diff
@@ -1,11 +1,13 @@
 // Spotify app client ID
-export const ClientId = '7fca87b90c354988acbffcb363ed7a60';
+export const ClientId = process.env.CLIENT_ID;
```
Comment (JeffreyCA) [inline utils/Constants.ts]: Can you please add these new env variables to the README? Under `#deploying-own-vercel-instance` and `#running-locally`. Thanks for making this change!

**Review by JeffreyCA — CHANGES_REQUESTED** [review]
Thank you for submitting these changes! 😃
I had a few comments

File: utils/Constants.ts (hunk 3 above)
Comment (setoelkahfi) [reply utils/Constants.ts]: I think it's already there. It's the `CLIENT_ID`. I just use the `.env` variables, which don't work currently. I keep getting `undefined` for everything in the `process.env.`

File: pages/api/index.ts (hunk 2 above)
Comment (setoelkahfi) [reply pages/api/index.ts]: Do you mean` recentlyPlayedItems = [...recentlyPlayedItems.slice(0, count)];`? Not familiar with JS :)

Comment (JeffreyCA) [reply pages/api/index.ts]: `recentlyPlayedItems = recentlyPlayedItems.slice(0, count);` should work

File: utils/Constants.ts (hunk 3 above)
Comment (JeffreyCA) [reply utils/Constants.ts]: I meant `BASE_URL` and `REDIRECT_URI`

File: pages/api/index.ts (hunk 2 above)
Comment (setoelkahfi) [reply pages/api/index.ts]: @JeffreyCA I pushed some updates :)

File: utils/Constants.ts (hunk 3 above)
Comment (JeffreyCA) [reply utils/Constants.ts]: Thank you :)

File: pages/api/index.ts (hunk 2 above)
Comment (JeffreyCA) [reply pages/api/index.ts]: Thanks!

**Comment by JeffreyCA** [comment]
@setoelkahfi Unfortunately the env var changes aren't working. The environment variables are server-side only so when you click authorize (client-side process), `client_id` and `redirect_uri` are undefined.

If you take a look [here](https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser), to expose the variables to the browser you'll need to append `NEXT_PUBLIC_` to those variable names. Can you make those changes?

Deployment: https://spotify-recently-played-readme-jdyo4ecyn-jeffreyca.vercel.app/

**Comment by setoelkahfi** [comment]
> @setoelkahfi Unfortunately the env var changes aren't working. The environment variables are server-side only so when you click authorize (client-side process), `client_id` and `redirect_uri` are undefined.
> 
> If you take a look [here](https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser), to expose the variables to the browser you'll need to append `NEXT_PUBLIC_` to those variable names. Can you make those changes?
> 
> Deployment: https://spotify-recently-played-readme-jdyo4ecyn-jeffreyca.vercel.app/

Yeah, that's what I said. I thought it's just me. I'll push the update.

**Comment by setoelkahfi** [comment]
@JeffreyCA I pushed the update. I think I tried it (prefix them with `NEXT_PUBLIC_`) before and had no luck. Haven't tried it again since then.

**Comment by JeffreyCA** [comment]
> @JeffreyCA I pushed the update. I think I tried it (prefix them with `NEXT_PUBLIC_`) before and had no luck. Haven't tried it again since then.

Thanks, you'll need to update this too: https://github.com/JeffreyCA/spotify-recently-played-readme/blob/0235a5c05204aa3ac0c3f05fb4790e28e8d431a2/utils/SpotifyAuthUtil.ts#L14

File: README.md [hunk 4]
```diff
@@ -57,13 +57,27 @@ Example:
...
+
+Example:
+```md
+![Spotify recently played](https://spotify-recently-played-readme.vercel.app/api?user=jeffreyca16&unique=true)
+```
+
+![Spotify recently played](https://spotify-recently-played-readme.vercel.app/api?user=jeffreyca16&unique=true)
+
 ## Deploying own Vercel instance
 [![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https%3A%2F%2Fgithub.com%2FJeffreyCA%2Fspotify-recently-played-readme&env=CLIENT_ID,CLIENT_SECRET,FIREBASE_PROJECT_ID,FIREBASE_PRIVATE_KEY_B64,FIREBASE_CLIENT_EMAIL)
```
Comment (JeffreyCA) [inline README.md]:
```markdown
[![Deploy with Vercel](https://vercel.com/button)](https://vercel.com/new/git/external?repository-url=https%3A%2F%2Fgithub.com%2FJeffreyCA%2Fspotify-recently-played-readme&env=NEXT_PUBLIC_CLIENT_ID,NEXT_PUBLIC_BASE_URL,NEXT_PUBLIC_REDIRECT_URI,CLIENT_SECRET,FIREBASE_PROJECT_ID,FIREBASE_PRIVATE_KEY_B64,FIREBASE_CLIENT_EMAIL)
```

File: README.md [hunk 5]
```diff
@@ -57,13 +57,27 @@ Example:
...
 
 Deploy your own Vercel instance using the link above. Next, set the following environment variables:
 
 | Name | Description |
 |---|---|
+| `REDIRECT_URI` | Callback URI from Spotify |
```
Comment (JeffreyCA) [inline README.md]: Need to update these variable names here, as well as in .env.example

**Review by JeffreyCA — COMMENTED** [review]
Thanks, the new dev deployment seems to be working now

File: README.md (hunk 5 above)
Comment (setoelkahfi) [reply README.md]: Done.

Comment (JeffreyCA) [reply README.md]: Thank you!

**Pull request merged**